security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,34 @@
+package crypter
+
+import (
+	"errors"
+)
+
+// counter is a 64-bit counter.
+type counter struct {
+	value         uint64
+	hasOverflowed bool
+}
+
+func newCounter(value uint64) counter {
+	return counter{value: value}
+}
+
+// getAndIncrement returns the current value of the counter and increments it.
+func (c *counter) getAndIncrement() (uint64, error) {
```

Comment:
I think @matthewstevenson88 had some thoughts on this. The initial PR separated these 2 APIs but we decided to combine them into one to avoid misusing the API.

Reply:
I suggested this change because we never want a sequence value to be reused (this would result in the same nonce being used to encrypt/decrypt two different records). To guarantee this, this function changes the sequence number whenever it is accessed. WDYT?